syntax = "proto3";

package pipeline;

service Pipeline {
  rpc Submit (SubmitRequest) returns (SubmitReply) {}
}

message SubmitRequest {
  string pipeline_spec = 1;
  repeated Input inputs = 2;
}

message SubmitReply {
  bool success = 1;
  string mpd_url = 2;
  string error_msg = 3;
}

message Input {
  string type = 1;
  string value = 2;
}
